sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -379,6 +379,31 @@ case class ArrayFilter(
   override def prettyName: String = "filter"
 }
 
+trait ArrayExistsForAllBase
+extends ArrayBasedSimpleHigherOrderFunction with CodegenFallback {
+  def check(cond: Boolean): Boolean
+
+  override def dataType: DataType = BooleanType
+  override def functionType: AbstractDataType = BooleanType
+
+  @transient lazy val LambdaFunction(_, Seq(elementVar: NamedLambdaVariable), _) = function
+
+  override def nullSafeEval(inputRow: InternalRow, argumentValue: Any): Any = {
+    val arr = argumentValue.asInstanceOf[ArrayData]
+    val f = functionForEval
+    var continue = true
+    var i = 0
+    while (i < arr.numElements && continue) {
+      elementVar.value.set(arr.get(i, elementVar.dataType))
+      if (check(f.eval(inputRow).asInstanceOf[Boolean])) {
+        continue = !continue
+      }
+      i += 1
+    }
+    !check(continue)
+  }
```

Comment:
I think the name for continue makes sense here, but I can see that the final `!check(continue)` returned value may be confusing to read. Anyone have suggestions to make the intent more clear here?

Reply:
What is the intent of the `!check(continue)` ?

Reply:
The old exists implementation:

Reply:
The previous exists implementation:
```scala
  override def nullSafeEval(inputRow: InternalRow, argumentValue: Any): Any = {
    val arr = argumentValue.asInstanceOf[ArrayData]
    val f = functionForEval
    var exists = false
    var i = 0
    while (i < arr.numElements && !exists) {
      elementVar.value.set(arr.get(i, elementVar.dataType))
      if (f.eval(inputRow).asInstanceOf[Boolean]) {
        exists = true
      }
      i += 1
    }
    exists
  }
```

Compare it to the first forall implementation I proposed:
```scala
  override def nullSafeEval(inputRow: InternalRow, argumentValue: Any): Any = {
    val arr = argumentValue.asInstanceOf[ArrayData]
    val f = functionForEval
    var forall = true
    var i = 0
    while (i < arr.numElements && forall) {
      elementVar.value.set(arr.get(i, elementVar.dataType))
      if (!f.eval(inputRow).asInstanceOf[Boolean]) {
        forall = false
      }
      i += 1
    }
    forall
  }

The two implementaitons were really similar and were abstracted into the current one:
```scala
  override def nullSafeEval(inputRow: InternalRow, argumentValue: Any): Any = {
    val arr = argumentValue.asInstanceOf[ArrayData]
    val f = functionForEval
    var continue = true
    var i = 0
    while (i < arr.numElements && continue) {
      elementVar.value.set(arr.get(i, elementVar.dataType))
      if (check(f.eval(inputRow).asInstanceOf[Boolean])) {
        continue = !continue
      }
      i += 1
    }
    !check(continue)
  }
```